lively.ide/studio/tabs.cp.js

```diff
@@ -198,15 +198,15 @@ class TabModel extends ViewModel {
   }
 
   setAppearance (isSelected) {
-    this.view.master = {};
+    let newMaster = {};
     if (isSelected) {
-      this.view.master.auto = this.selectedTabMaster;
+      newMaster.auto = this.selectedTabMaster;
     } else {
-      this.view.master.auto = this.defaultTabMaster;
+      newMaster.auto = this.defaultTabMaster;
     }
-    this.view.master.hover = isSelected ? false : this.hoveredTabMaster;
-    this.view.master.click = this.clickedTabMaster;
-    this.view.requestMasterStyling();
+    if (!isSelected) newMaster.hover = this.hoveredTabMaster;
+    newMaster.click = this.clickedTabMaster;
+    this.view.master = newMaster;
```

Comment:
That's the thing. It was actually never really supported to change masters like `aMorph.master.hover = HoverMaster` you always had to do this by assigning master a completely new value, like so: `aMorph.master = { hover: HoverMaster }` . This is in tune with how we handle other properties such as `position` or `extent`. Here is is also not supported to do the `aMorph.position.x = newWidth;` etc. It may `work` but only by the renderer accidentally traversing the morph due to some other reason.